DataFormats/EgammaReco/interface/EgTrigSumObj.h

```diff
@@ -0,0 +1,68 @@
+#ifndef DataFormats_EgammaReco_EgTrigSumObj_h
+#define DataFormats_EgammaReco_EgTrigSumObj_h
+
+#include "DataFormats/EgammaReco/interface/SuperCluster.h"
+#include "DataFormats/EgammaReco/interface/ElectronSeed.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+
+#include "DataFormats/EgammaReco/interface/SuperClusterFwd.h"
+#include "DataFormats/EgammaReco/interface/ElectronSeedFwd.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrackFwd.h"
+
+#include <unordered_map>
+#include <string>
+
+namespace reco {
+  class RecoEcalCandidate;
+}
+
+namespace reco {
+  class EgTrigSumObj {
```

Comment:
Perhaps `EgammaTriggerSummary`?
(just to be more readable and not go to even shorter `EgTrSumOb`)

Reply:
I would prefer EgTrigSumObj as EgammaTriggerSummary suggests its a summary of the entire trigger rather than a specific e/gamma object.  I could also go to EgHLTSumObj  or EgHLTObj. However its not something I feel super strong.

Reply:
`EgammaHLTSummaryObject` ?

Reply:
could do, I mean I'm trying to have a relatively short name. I think "EgHLT" should be an acceptable abbreviation which anybody using this class will figure out. I do concede on reflection "Sum" is a bit less clear. Obj I would imagine is also fairly clear.  

With the summary, I'm trying to get across that its an object which is produced at the end of the HLT rather than an object which is part of the HLT decision process.  Now this is important to get right as there will be very soon be a MuHLTSumObj, a TauHLTSumObj, a BTagHLTSumObj and a JetHLTSumObj (although jet & btag will likely merge).  

So I do appreciate suggestions here, I know I'm terrible at names.  

I could live with 
EgHLTSummaryObject 
MuHLTSummaryObject
TauHLTSummaryObject
JetHLTSummaryObject

but I would prefer something a bit shorter but not to the point thats confusing. 

Thoughts?

Reply:
I'd still use Muon and Egamma, these are common class naming patterns in CMSSW and it's better to be consistent.

> With the summary, I'm trying to get across that its an object which is produced at the end of the HLT rather than an object which is part of the HLT decision process.

Looking at existing patterns,
isn't a `TriggerObject` from DataFormats/HLTReco/interface/TriggerObject.h supposed to cover the final objects as well?
Then, `EgammaTriggerObject` seems like a natural extension.

But if the endpoint needs to be included, berhaps `EgammaTriggerResultObject` ?

given the trigger/HLT specificity, does this have to be in the `reco` namespace?

Also, given the ideas for other variants, wouldn't it be better to inherit from smth like `TriggerObject` for some common interface?

Reply:
... also, if the `trigger` namespace is acceptable, the `Trigger` can be dropped from the class name.

Reply:
apparently https://github.com/cms-sw/cmssw/pull/32486#discussion_r554024526
and https://github.com/cms-sw/cmssw/pull/32486#discussion_r554025282 were ignored.
It would be nice to at least have a comment why they were not considered.

Reply:
sorry they were not ignored, just missed (my bad, I need to fix my email filters better.)

Reply:
> apparently [#32486 (comment)](https://github.com/cms-sw/cmssw/pull/32486#discussion_r554024526)
> and [#32486 (comment)](https://github.com/cms-sw/cmssw/pull/32486#discussion_r554025282) were ignored.
> It would be nice to at least have a comment why they were not considered.

Sorry Slava, I also missed them during the review.
Removing the "+1" while waiting for Sam making the update

Reply:
this indeed might be better.

What do you think @fwyzard, @Martin-Grunewald

Reply:
EgammaTriggerObject is fine with me especially if you extend TriggerObject.